cli/azd/pkg/auth/constants.go

```diff
@@ -0,0 +1,15 @@
+package auth
+
+import "fmt"
+
+const (
+	//#nosec G101 -- This is a false positive
+	remoteAuthCredName string = "RemoteCredential"
+
+	contentTypeHeader   string = "Content-Type"
+	authorizationHeader string = "Authorization"
```

Comment:
I wonder if that these constants may end up adding overhead. There's overhead where we would have to lookup these values in the "constants library" to provide the value, instead of just typing "Content-Type" and "Authorization". These constants are not subjected to change since they're cemented in the HTTP standards, so refactoring here doesn't give us the added benefit of central control we get with most constants.

When implementing a client request, it's likely that you're reading the spec from the API, which would also express the header keys directly.

On the flip side, if the idea is really "we want to make writing better", something like a `headers` library that exposes these keys would make it slightly nicer. I'd still be slightly opposed to it, as I always really appreciate the aspect of `go` of not exposing more abstractions than necessary.

Reply:
Yeah, I wouldn't create a new package just for http abstractions. The intention of having string constants (not even a new type, but just the string) within a package is usually:

- Easy to update (as you mentioned correctly, this will never change)
- Prevent typos when you are adding the string manually (you might ended up doing a lookup in the code or in the web to make sure how the to spell the header right. So you don't write `contentType` or `content_type`). Copilot should make even easier, finding an existing constant in the package for what you need.
- Perf.  Might take a few less cycles for the compiler to find and handle those repeated constants.